packages/apm/src/integrations/tracing.ts

```diff
@@ -629,6 +648,69 @@ export class Tracing implements Integration {
     // tslint:enable: no-unsafe-any
   }
 
+  /**
+   * Starts tracking the Largest Contentful Paint on the current page.
+   */
+  private static _trackLCP(): void {
+    // Based on reference implementation from https://web.dev/lcp/#measure-lcp-in-javascript.
+
+    // Use a try/catch instead of feature detecting `largest-contentful-paint`
+    // support, since some browsers throw when using the new `type` option.
+    // https://bugs.webkit.org/show_bug.cgi?id=209216
+    try {
+      // Keep track of whether (and when) the page was first hidden, see:
+      // https://github.com/w3c/page-visibility/issues/29
+      // NOTE: ideally this check would be performed in the document <head>
+      // to avoid cases where the visibility state changes before this code runs.
+      let firstHiddenTime = document.visibilityState === 'hidden' ? 0 : Infinity;
+      document.addEventListener(
+        'visibilitychange',
+        event => {
+          firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
+        },
+        { once: true },
+      );
+
+      const updateLCP = (entry: PerformanceEntry) => {
+        // Only include an LCP entry if the page wasn't hidden prior to
+        // the entry being dispatched. This typically happens when a page is
+        // loaded in a background tab.
+        if (entry.startTime < firstHiddenTime) {
+          // NOTE: the `startTime` value is a getter that returns the entry's
+          // `renderTime` value, if available, or its `loadTime` value otherwise.
+          // The `renderTime` value may not be available if the element is an image
+          // that's loaded cross-origin without the `Timing-Allow-Origin` header.
+          Tracing._lcp = {
+            // @ts-ignore
+            elementId: entry.id,
+            // @ts-ignore
+            elementSize: entry.size,
+            largestContentfulPaint: entry.startTime,
+          };
```

Comment:
`entry.id` and `entry.size` are not always available.
We may or may not want more details from the LCP entry.
I've put those here as examples of what is possible.

https://developer.mozilla.org/en-US/docs/Web/API/LargestContentfulPaint

@HazAT @kamilogorek @dashed feel free to comment on this.

Also note that we're only sending a number like `63ms` instead of a UNIX timestamp (like in spans).

Reply:
I think `entry.startTime` would be relative to `performance.timeOrigin`. 

Is it sufficient to infer the start timestamp relative to the transaction's start timestamp? 

This is assuming that the transaction's start timestamp uses `performance.timeOrigin` in some fashion.

Reply:
@dashed yes, yes, yes :)

Since the LCP metric is defined in terms of a page load, when looking at aggregation reporting we should probably look at it in isolation from transaction start timestamps.

For overlaying into a single `pageload` transaction view (a.k.a. vertical line), it should be okay-ish to consider it an offset from the transaction start timestamp.